src/help/utils/apiSnippets.tsx

```diff
@@ -0,0 +1,312 @@
+        2
+      )});`
+    : ''
+}
+const url = \`${url.origin + url.pathname}${searchParams ? `?$\{searchParams}` : ''}\`;
+
+const response = await fetch(url, {
+  headers: ${indent(JSON.stringify(Object.fromEntries(headers.entries()), null, 2))},${
+    method === 'GET'
+      ? ''
+      : `
+  method: "${method}",`
+  }${
+    body
+      ? `
+  body: new FormData(${indent(JSON.stringify(body, null, 2))}),`
+      : ''
+  }
+});
+if (!response.ok) {
+  throw new Error(\`$\{response.status}: $\{response.statusText}\`);
+}
+
+const data = await response.${isJSON ? 'json' : 'text'}();
+
+console.log(data);
+`;
+    },
+  },
+};
+
+// PERL
+const SnippetGeneratorPerlPlugin: SnippetGeneratorPlugin = {
+  fn: {
+    // use `requestSnippetGenerator_` + key from config (node_native) for generator fn
+    requestSnippetGenerator_perl: (request) => {
+      const { url, method, headers, isJSON } = requestExtract(request);
+
+      if (method === 'POST') {
+        return '# No automatic snippet, work in progress';
+      }
+
+      return `# Perl not recommended, limited support
+# Make sure to have installed the below libraries
+# cpanm Type::Tiny IO::Socket::SSL ${isJSON ? ' JSON' : ''}
+use strict;
+use warnings;
+use HTTP::Tiny;${isJSON ? '\nuse JSON;' : ''}
+
+my $http = HTTP::Tiny->new();
+
+my $url = '${url}';
+my $response = $http->get($url, {
+    headers => {
+${Array.from(headers.entries())
+  .map(
+    ([key, value], index) =>
+      `${index === 0 ? '' : ',\n'}        '${key}' => '${value}'`
+  )
+  .join('')}
+    }
+});
+
+if (!$response->{success}) {
+    print "Error " . $response->{status} . ": " . $response->{reason} . "\n";
+    print "Response content: " . $response->{content} . "\n";
+    die;
+}
+
+${
+  isJSON
+    ? `my $content = decode_json($response->{content});
+use Data::Dumper;
+print Dumper($content);`
+    : `my $content = $response->{content};
+print $content;`
+}
+`;
+    },
+  },
+};
+
+// R
+const SnippetGeneratorRPlugin: SnippetGeneratorPlugin = {
+  fn: {
+    // use `requestSnippetGenerator_` + key from config (node_native) for generator fn
+    requestSnippetGenerator_r: (request) => {
+      const { url, searchParams, method, headers, isJSON } =
+        requestExtract(request);
+
+      if (method === 'POST') {
+        return '# No automatic snippet, work in progress';
+      }
+
+      return `# R not recommended, limited support
+# Install the required package if not already installed
+# install.packages("httr")${isJSON ? '\n# install.packages("jsonlite")' : ''}
+
+library(httr)${isJSON ? '\nlibrary(jsonlite)' : ''}
+
+base_url <- "${url.origin + url.pathname}"${
+        searchParams
+          ? `
+params <- list(
+${Array.from(searchParams.entries())
+  .map(
+    ([key, value], index) => `${index === 0 ? '' : ',\n'}  ${key} = "${value}"`
+  )
+  .join('')}
+)`
+          : ''
+      }
+
+response <- GET(
+  base_url,${
+    searchParams
+      ? `
+  query = params,`
+      : ''
+  }
+  add_headers(
```

Comment:
I didn't look further than `httr` because I looked at the Proteins API snippets and it was using that, but I guess using `httr2` would be better.
Should I let you port the snippet from your comment into the code or do you prefer I do it?

Reply:
I'm happy to port over to httr2 at some point today.